### ISSUE TITLE
Format string is hard-coded to be a string literal

### ISSUE THREAD

**Issue by gh-andre** [MengRao/fmtlog#27]
If the format string is not a string literal, cryptic compiler messages are reported and at first it's not even clear what the problem is. For example:
```c++
template <typename ... T>
void mylog(const std::string_view& format, T&& ... arg)
{
  FMTLOG_ONCE(fmtlog::INF, format, std::forward<T>(arg)...);
}

int main(void)
{
  mylog("test log line {} {}"sv, "ABC", 123);
  return 0;
}
```
GCC reports `use of parameter from containing function` against the `format` parameter and an additional error `'constexpr` call flows off the end of the function` in `core.h` in `fmt`. VC++ reports just `function cannot access 'fmt'`.

This is caused by `FMT_STRING` used in all logging macros, which only works with string literals. Removing `FMT_STRING` from logging macros fixes these errors.

**Comment by MengRao** [comment]
Currently only literally format string is supported, you can wrap the log statements using macros simply as `logio` does.

**Comment by gh-andre** [comment]
Yes, that's what I ended up doing - I split `FMTLOG_ONCE` to allow location supplied as an argument, so it can be propagated through the application, until it reaches `fmtlog`.

You may want to document in README.md that only string literals are supported for formats. This will make it more clear that for applications that would need to wrap the logger in their own layer it won't work without patching.

Closing this issue, since you consider this working as designed. I will update my v2.1.1 patch at some point for those who's interested in this path.

**Comment by gh-andre** [comment]
Implemented the change mentioned above, which allows format in `FMTLOG_ONCE` to be any string type and also supply source location explicitly, if anyone is interested (commit https://github.com/gh-andre/fmtlog/commit/bd68c0bd30d8ed830da14c73abfa8d7cec3b5498) on `v2-1-1-patches` branch.

https://github.com/gh-andre/fmtlog/tree/v2-1-1-patches

**Comment by MengRao** [comment]
Please check latest update, now FMT_STRING is removed and the format string type is the same as fmt::format/print.

**Comment by gh-andre** [comment]
Thanks. I merged my changes with the new code. Here's the branch with all changes, if you want to consider any of it.

https://github.com/gh-andre/fmtlog/tree/main-patches
